numpy/core/src/multiarray/ctors.c

```diff
@@ -2005,9 +2011,31 @@ PyArray_FromArray(PyArrayObject *arr, PyArray_Descr *newtype, int flags)
             return NULL;
         }
 
-        if (flags & NPY_ARRAY_UPDATEIFCOPY)  {
+        if (flags & NPY_ARRAY_UPDATEIFCOPY) {
+            /* This is the ONLY place the NPY-ARRAY_UPDATEIFCOPY flag
+             * is checked to call PyArray_SetUpdateIfCopyBase
+             * Deprecate this (and remove the `else` from the following
+             * condition) once the flag itself is deprecated
+             */
+
```

Comment:
This wording is a bit confusing. A more accurate statement would be "this logic branch is deprecated as of version X with date."